staging/src/k8s.io/client-go/transport/transport.go

```diff
@@ -52,7 +52,7 @@ func New(config *Config) (http.RoundTripper, error) {
 // TLSConfigFor returns a tls.Config that will provide the transport level security defined
 // by the provided Config. Will return nil if no transport level security is requested.
 func TLSConfigFor(c *Config) (*tls.Config, error) {
-	if !(c.HasCA() || c.HasCertAuth() || c.TLS.Insecure) {
+	if !(c.HasCA() || c.HasCertAuth() || c.TLS.Insecure || len(c.TLS.ServerName) > 0) {
```

Comment:
if tls.Insecure is true, why don't we just return nil, nil?

Reply:
anything that deviates from the defaults must return a custom tls.Config

setting Insecure=true or a custom ServerName requires a non-default tls.Config